packages/magento-customer/components/GuestOrderOverview/GuestOrderOverviewForm.tsx

```diff
@@ -89,6 +90,7 @@ export function GuestOrderOverviewForm() {
           <Typography sx={(theme) => ({ textAlign: 'center', mb: theme.spacings.lg })}>
             <OrderStateLabel items={orderData?.guestOrder.items} />
           </Typography>
+          <OrderActionButtons {...orderData?.guestOrder} />
```

Comment:
This would have only shown the reorder button in this case. Though after testing it seems only customer orders can actually be reordered so I removed it here.